pkg/keys/printer.go

```diff
@@ -748,3 +714,32 @@ func PrettyPrintRange(start, end roachpb.Key, maxChars int) string {
 
 	return b.String()
 }
+
+// copyEscape copies the string to the buffer, and avoids writing
+// invalid UTF-8 sequences and control characters.
+func copyEscape(buf *bytes.Buffer, s string) {
+	buf.Grow(len(s))
+	// k is the index in s before which characters have already
+	// been copied into buf.
+	k := 0
+	for i := 0; i < len(s); i++ {
+		c := s[i]
+		if c < utf8.RuneSelf && strconv.IsPrint(rune(c)) {
+			continue
+		}
+		buf.WriteString(s[k:i])
+		l, width := utf8.DecodeRuneInString(s[i:])
+		if l == utf8.RuneError || l < 0x20 {
+			const hex = "0123456789abcdef"
```

Comment:
any reason not to do something like `fmt.Sprintf("\\x%02x", c])` which we do below?